omod/src/main/java/org/openmrs/module/smartonfhir/page/controller/SmartAppsRedirectPageController.java

```diff
@@ -16,16 +16,22 @@
 public class SmartAppsRedirectPageController {
 	
 	public Redirect get(@RequestParam(value = "app") AppDescriptor app,
-	        @RequestParam(required = false, value = "patientId") String patientId) {
+	        @RequestParam(required = false, value = "patientId") String patientId,
+	        @RequestParam(required = false, value = "visitId") String visitId) {
 		
 		String launchUrl = app.getConfig().get("launchUrl").getTextValue();
 		
-		// For EHR launch
-		if (patientId != null) {
+		//EHR launch for Visit Context
+		if (!visitId.isEmpty()) {
+			return new Redirect("ms/smartVisitEhrLaunchServlet?launchUrl=" + launchUrl + "&visitId=" + visitId
+			        + "&patientId=" + patientId);
+		}
+		
+		// EHR launch for Patient Context
+		if (!patientId.isEmpty()) {
```

Comment:
A couple of things here:

1. Can we maybe send a parameter to indicate whether this is an EHR launch or standalone launch rather than trying to guess from the supplied parameters?
2. Why do we need two servlets for EHR launch?
3. Heck, why do we need a separate servlet for EHR launch or Standalone launch. Can't that logic be moved into a single servlet?

Reply:
Yes, we can add a field in the AppTemplate file, which indicates where it is an EHR or standalone launch. Moreover, we can also add one more field, which indicates the context of launch (Patient or Encounter) and redirect it to the single servlet for EHR. Thought?

Yes, we can use a single servlet for both EHR and standalone, but keeping the two different launch nature separate will be less confusing and easy to understand. Thought?

Reply:
I think one servlet with the ability to distinguish what it's doing based on a parameter is probably easier from a maintenance point of view then trying to figure out which of two similarly named servlets are responsible for which behaviour.

I like the idea of including a field in the AppTemplate; that definitely makes sense!

Reply:
@ibacher thanks